CMakePresets.json

```diff
@@ -0,0 +1,104 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "generator": "Ninja",
+      "binaryDir": "${sourceDir}/build",
+      "cacheVariables": {
+        "CMAKE_EXPORT_COMPILE_COMMANDS": true,
+        "DEPS_PREFIX": "${sourceDir}/.deps/usr",
+        "PREFER_LUA": false,
+        "COMPILE_LUA": true
+      },
```

Comment:
Hmm, it doesn't really make sense to me to specify options if it uses the default value anyway and if it's not required for the presets to work.

Reply:
It's more about showing _configurable_ defaults here, as one of the nice benefits of these presets is discoverability.

I can still move them to the example `CMakeUserPresets.json`, but I figured these deserved better visibility.

Reply:
Nah, `ccmake` is the tool to discover options. This gives the false impression that the options is required for the preset to work.